internals/detect_intent.py

```diff
@@ -160,10 +160,9 @@ def is_lgtm_allowed(from_addr, feature, approval_field):
   return allowed
 
 
-def detect_new_thread(feature_id, approval_field):
-  """Return True if there are no previous approval values for this intent."""
-  existing_votes = Vote.get_votes(
-      feature_id=feature_id, gate_type=approval_field.field_id)
+def detect_new_thread(feature_id: int, gate_id: int) -> bool:
+  """Return True if there are no previous approval values for this gate."""
+  existing_votes = Vote.get_votes(feature_id=feature_id, gate_id=gate_id)
```

Comment:
This could just pass `gate_id=gate_id` because the gate also determines the feature.

Reply:
Good call! I've made the change. 🙂